core/client/fs/src/main/java/alluxio/client/file/cache/store/MemoryPageStore.java

```diff
@@ -94,4 +107,52 @@ public void close() {
   public void reset() {
     mPageStoreMap.clear();
   }
+
+  private class MemPage {
+    private final byte[] mPage;
+    private int mPageLength;
+
+    public MemPage(byte[] page, int pageLength) {
+      mPage = page;
+      mPageLength = pageLength;
+    }
+
+    public byte[] getPage() {
+      return mPage;
+    }
+
+    public int getPageLength() {
+      return mPageLength;
+    }
+
+    public void setPageLength(int pageLength) {
+      mPageLength = pageLength;
+    }
+  }
+
+  private class PagePool {
```

Comment:
is it possible that we have the same issue as the ByteBuffer pool that we release the buffer too early? Just want to double check